include/libtorrent/aux_/vector.hpp

```diff
@@ -52,14 +52,14 @@ namespace libtorrent { namespace aux {
 		{
 			TORRENT_ASSERT(idx >= IndexType(0));
 			TORRENT_ASSERT(idx < end_index());
-			return this->base::operator[](static_cast<underlying_index>(idx));
+			return this->base::operator[](static_cast<std::size_t>(idx));
```

Comment:
I don't think this will work. the conversion operator to the underlying type is explicit, you can only cast to that specific type. you could do two ``static_cast``s though